### DISCUSSION TITLE
How to load a color from currently selected color scheme during action callback

### DISCUSSION THREAD

**Discussion by brycewcole** [wezterm/wezterm#5242] [Q&A] [answered]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

Hello everyone, 

I am making a small script that helps me jump between different sessions. Im using some of the prebuilt UI elements to achieve this (InputSelector). It gets triggered by a hotkey and then executed in a action callback like 
```
	return wezterm.action_callback(function(window, pane)
```

I wanted to add a little bit of color to the text, but as someone who switches their colors scheme around a lot I also didnt want to hardcode that color. Is there a way to access the current color scheme and then pulll like the foreground or one of the bright colors from that?

Thanks in advance, really loving the terminal so far.

**Comment by wez** [comment, marked as answer]
Take a look at:
* https://wezfurlong.org/wezterm/config/lua/window/effective_config.html
* https://wezfurlong.org/wezterm/config/lua/wezterm.color/get_builtin_schemes.html

**Reply by brycewcole** [reply]
Thank you! I ended up using that to write a small helper function... have a great weekend!
```lua
local function getColorByKey(window, key)
	local current_scheme = wezterm.color.get_builtin_schemes()[window:effective_config().color_scheme]

	local key_to_color = {
		black = current_scheme.ansi[1],
		red = current_scheme.ansi[2],
		green = current_scheme.ansi[3],
		yellow = current_scheme.ansi[4],
		blue = current_scheme.ansi[5],
		magenta = current_scheme.ansi[6],
		cyan = current_scheme.ansi[7],
		white = current_scheme.ansi[8],
		black_bright = current_scheme.brights[1],
		red_bright = current_scheme.brights[2],
		green_bright = current_scheme.brights[3],
		yellow_bright = current_scheme.brights[4],
		blue_bright = current_scheme.brights[5],
		magenta_bright = current_scheme.brights[6],
		cyan_bright = current_scheme.brights[7],
		white_bright = current_scheme.brights[8],
	}

	return key_to_color[key]
end
```
